src/Common/recorder/JamRecorder.h

```diff
@@ -166,6 +166,8 @@ class JamMetadataWriter
 public:
     virtual void write(const Jam &metadata) = 0;
     virtual ~JamMetadataWriter(){}
+    virtual QString getWriterId() const {return "";}
```

Comment:
Code review: use virtual pure functions to avoid new inherited classes using wrong/empty IDs.